### PR TITLE
Feature/sass styles

### PR DESCRIPTION
**Pull request by kevinglover**
### Added Sass for styles

Support for Sass is growing. Annotator.js styles should be using the latest best practices in web development. Implementing Sass styles in Annotator will make it easier to integrate with other frameworks and libraries. 
- Sass files are converted from the existing CSS styles
- Uses Grunt.js to compile, minify, and copy the files to the right directory
- It's only optional for use; the compiled CSS are still usable without any extra effort
- Added minified styles and maps as well

### CODE REVIEW

**Comment by tilgovi** [comment]
This is great, thanks.

For development, it's really wonderful to be able to use the `serve` tool and for build we current have the `Makefile`. I would consider replacing the `Makefile` but not in this PR.

Would you mind removing the grunt portion of this and just adding a rule to build the scss source in the `Makefile` as well as a route for the CSS in `tools/serve`? You should be able to use `node-sass` from the CLI for the `Makefile` and from code for `tools/serve`.

**Comment by tilgovi** [comment]
Or if `make` isn't super familiar with you I'm happy to do this.

**Comment by nikolas** [comment]
I agree, the Gruntfile seems unnecessary. Instead, we can just add a `make css` rule or something to the makefile.

**Comment by kevinglover** [comment]
@tilgovi Does this look more like what you had in mind?

File: Makefile [hunk 1]
```diff
@@ -22,6 +24,16 @@ pkg/%.min.js: pkg/%.js
 	@echo Writing $@
 	@$(UGLIFYJS) --preamble "$$(tools/preamble)" $< >$@
 
+# compile Sass
```
Comment (tilgovi) [inline Makefile]: These look great, but we can add the dependencies and get smart recompiling.

File: Makefile [hunk 2]
```diff
@@ -22,6 +24,16 @@ pkg/%.min.js: pkg/%.js
...
+sass:
```
Comment (tilgovi) [inline Makefile]: For instance, here, we could list the dependencies after the colon. That way, it will only recompile when necessary.

**Comment by tilgovi** [comment]
I started to comment with specific changes, but it's hard to communicate all the make specifics over comments. Would you be already if I took this into a branch and added some fixups to the Makefile stuff, then you can look it over and try it out before I merge it?

**Comment by kevinglover** [comment]
@tilgovi Sure thing!

**Comment by tilgovi** [comment]
Took me longer to get back to this than I wanted but I was looking at it today. I've almost got it ready but making this change breaks the browserify bundle that is currently trying to use insert-css. I'm really not a fan of inlining CSS into a JS bundle so I think I'll unbundle that first. Then, I've got the Makefile cleanup ready to go.

**Pull request closed without merging**